lib/xray_creator.py

```diff
@@ -47,10 +47,7 @@ def _initialize_books(self):
         book_lookup = {}
         for book in self.books_not_failing():
             self._total_not_failing += 1
-            if book_lookup.has_key(book.title):
```

Comment:
I think this should work.  To be complete, I think the change is from this:
```
        book_lookup = {}
        for book in self.books_not_failing():
            self._total_not_failing += 1
            if book_lookup.has_key(book.title):
                book_lookup[book.title].append(book)
                continue
            book_lookup[book.title] = [book]
```
to this:
```
        book_lookup = defaultdict{list}
        for book in self.books_not_failing():
            self._total_not_failing += 1
            book_lookup[book.title].append(book)
```
The fact self.books_not_failing is a genarator shouldn't matter - the loop handles one `book` at a time.  Obviously if I'm wrong then say, always happy to learn :)